File: P133B.cpp
#include <iostream>
#include <cmath>
#include <algorithm>
#include <vector>
#include <string>

#define ll long long

using namespace std;

/*
* B - Unary
*/

string toBinary(string unary);
ll getAnswer(string binary);

int main()
{
	string bf;
	cin >> bf;

	cout << getAnswer(toBinary(bf));

	return 0;
}

string toBinary(string unary)
{
	string binary;

	for(int i = 0, len = unary.length(); i < len; i++)
	{
		if(unary[i] == '>') binary += "1000";
		else if(unary[i] == '<') binary += "1001";
		else if(unary[i] == '+') binary += "1010";
		else if(unary[i] == '-') binary += "1011";
		else if(unary[i] == '.') binary += "1100";
		else if(unary[i] == ',') binary += "1101";
		else if(unary[i] == '[') binary += "1110";
		else if(unary[i] == ']') binary += "1111";
	}

	return binary;
}

ll getAnswer(string binary)
{
	ll answer = 0;

	for(int i = 0, len = binary.length(); i < len; i++) answer = (answer * 2 + (binary[i] - '0')) % 1000003;

	return answer;
}

File: P133B.java
import java.util.Scanner;

/*
* A - Unary
*/
 
public class P133B
{
  public static void main(String[] args)
  {
    Scanner scanner = new Scanner(System.in);
    String input = scanner.nextLine();
    scanner.close();
    
    System.out.println(getAnswer(toBinary(input)));
  }
    
  static String toBinary(String unary)
  {
	  String result = "";
	    for(byte i = 0; i < unary.length(); i++)
	    {
	      if(unary.charAt(i) == '>') result += "1000";
	      else if(unary.charAt(i) == '<') result += "1001";
	      else if(unary.charAt(i) == '+') result += "1010";
	      else if(unary.charAt(i) == '-') result += "1011";
	      else if(unary.charAt(i) == '.') result += "1100";
	      else if(unary.charAt(i) == ',') result += "1101";
	      else if(unary.charAt(i) == '[') result += "1110";
	      else if(unary.charAt(i) == ']') result += "1111";
	    }
	    
	    return result;
  }
  static long getAnswer(String binary)
  {
	  long answer = 0;
      for(int i = 0; i < binary.length(); i++) answer = (answer * 2 + (binary.charAt(i) - '0')) % 1000003;
      
      return answer;
   }
}